test/dartdoc_options_test.dart

```diff
@@ -43,21 +120,390 @@ void main() {
+  parentOverride: 'parent'
+  dirOption: 'firstSub'
+  fileOptionList: ['existing.dart', 'thing/that/does/not/exist']
+  mySpecialInteger: 11
+  fileOption: "not existing"
         ''');
     dartdocOptionsTwoFirstSub.writeAsStringSync('''
 dartdoc:
   categoryOrder: ['options_two_first_sub']
+  parentOverride: 'child'
+  nonCriticalDirOption: 'not_existing'
     ''');
   });
 
   tearDownAll(() {
     tempDir.deleteSync(recursive: true);
   });
 
+  group('new style dartdoc both file and argument options', () {
+    test(
+        'validate argument with wrong file throws error even if dartdoc_options is right',
+        () {
+      dartdocOptionSetBoth
+          .parseArguments(['--file-option', 'override-not-existing.dart']);
+      String errorMessage;
+      try {
+        dartdocOptionSetBoth['fileOption'].valueAt(firstDir);
+      } on DartdocFileMissing catch (e) {
+        errorMessage = e.message;
+      }
+      expect(
+          errorMessage,
+          equals(
+              'Argument --file-option, set to override-not-existing.dart, resolves to missing path: "${pathLib.join(pathLib.canonicalize(Directory.current.path), "override-not-existing.dart")}"'));
+    });
+
+    test('validate argument can override missing file', () {
+      dartdocOptionSetBoth.parseArguments(
+          ['--file-option', pathLib.canonicalize(firstExisting.path)]);
+      expect(dartdocOptionSetBoth['fileOption'].valueAt(secondDir),
+          equals(pathLib.canonicalize(firstExisting.path)));
+    });
+
+    test('File errors still get passed through', () {
+      dartdocOptionSetBoth.parseArguments([]);
+      String errorMessage;
+      try {
+        dartdocOptionSetBoth['fileOption'].valueAt(secondDir);
+      } on DartdocFileMissing catch (e) {
+        errorMessage = e.message;
+      }
+      expect(
+          errorMessage,
+          equals(
+              'Field dartdoc.fileOption from ${pathLib.canonicalize(dartdocOptionsTwo.path)}, set to not existing, resolves to missing path: '
+              '"${pathLib.join(pathLib.canonicalize(secondDir.path), "not existing")}"'));
+    });
+
+    test('validate override behavior basic', () {
+      dartdocOptionSetBoth.parseArguments(
+          ['--not-in-any-file', 'aha', '--map-option', 'over::theSea']);
+      expect(dartdocOptionSetBoth['mapOption'].valueAt(tempDir),
+          equals({'over': 'theSea'}));
+      expect(dartdocOptionSetBoth['mapOption'].valueAt(firstDir),
+          equals({'over': 'theSea'}));
+      expect(dartdocOptionSetBoth['notInAnyFile'].valueAt(firstDir),
+          equals('aha'));
+      expect(dartdocOptionSetBoth['mySpecialInteger'].valueAt(firstDir),
+          equals(30));
+    });
+
+    test('validate override behavior for parent directories', () {
+      dartdocOptionSetBoth.parseArguments(['--my-special-integer', '14']);
+      expect(
+          dartdocOptionSetBoth['mySpecialInteger'].valueAt(secondDirFirstSub),
+          equals(14));
```

Comment:
I think it's just a style thing but I prefer to elide explicit calls to equals since I think it defaults to that matcher.  In other words, you can just say `14`. Totally up to you though!

Reply:
general dartdoc style is to be explicit in tests, so I've continued that here.